src/twisted/logger/test/test_capture.py

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+
+"""
+Test cases for L{twisted.logger._capture}.
+"""
+
+from twisted.logger import Logger, LogLevel
+from twisted.trial.unittest import TestCase
+
+from .._capture import capturedLogs
```

Comment:
maybe the canonical import should be used here, as a way to exemplify how we expect it to be publicly available. 

so this can look like

```
from twisted.logger import capturedLogs
```

is there any advantage of using the relative imports?

Reply:
I used relative imports all over the logger package, so I think this should stay for consistency unless we wanna redo that usage in the package.

The advantage was that that module started in a different place and moving it into Twisted was a lot easier to manage with fewer import rewrites.

I do see you point, though, in that we are testing the private namespace and not the public namespace… I wouldn't object to a separate ticket for redoing that in test cases, I guess.

Reply:
I am not a big fan of separate tickets as most of the time nothing will happen.

Instead, I prefer drive-by fixes based on "The Boy Scout Rule" http://www.informit.com/articles/article.aspx?p=1235624&seqNum=6

Reply:
It's tempting to try to include random drive-by fixes in nearby code while you're maintaining something unrelated.  I've certainly done it.  But I think it's an antipattern.

Why is it tempting?

It's tempting because the overhead of an individual PR is too high, and so we're hoping to get more stuff through given each "cycle" of the code-review machinery, to reduce that overhead.

Why is the overhead of the PR review cycle high?

1. Fixed overhead, such as CI runtimes.  (We should work to decrease these, of course.)
2. Reviewer bandwidth; there's limited bandwidth for reviewers to review things, and so they can't spend that much time reviewing.
3. Reviewer latency; reviewers don't necessarily have time to come back every day.

Attempting to fix unrelated stuff allows you to reduce (1) (more changes per CI run) while *increasing* the overhead associated with 2 and 3.

By making bigger changes, you take up more of the time available to reviewers to evaluate your changes in each sitting, which (a) decreases the number of changes they can review, and (b) makes it less likely that they'll be able to come back again promptly to re-review the changes, both because you took up more of their time at a go, and you required them to think harder and generally use up more mental energy.

Make every change as small and self-contained and obvious as possible, and a reviewer with a few idle minutes can go in and stamp it, and possibly review multiple tickets in a sitting.

Reply:
thanks for the explanation.
Still, I was adding a comment to a new code which was written... why write it in one way and then change it in a separate PR, when you can write it from the start

I am not talking about rewriting all the imports.

Reply:
I do think there's something to be said for switching from one consistent style to another consistent style, but - point taken ;).